src/components/ifxRange/ifxRange.tsx

```diff
@@ -0,0 +1,85 @@
+import { Component, h, Prop, Event, EventEmitter } from '@stencil/core';
+
+@Component({
+  tag: 'ifx-range',
+  styleUrl: 'ifxRange.scss',
+  shadow: true,
+})
+export class IfxRange {
+  @Prop() min: number = 0;
+  @Prop() max: number = 100;
+  @Prop({ mutable: true }) value: number = 0;
+  @Prop() disabled: boolean = false;
+  @Prop() showPercentage: boolean = false; // New property for toggleable percentage display
+  @Prop() leftIcon: string;
+  @Prop() rightIcon: string;
+  @Prop() leftText: string;
+  @Prop() rightText: string;
+  @Event() valueChanged: EventEmitter<number>;
+
+  private inputRef: HTMLInputElement;
+
+  handleInputChange(event: Event) {
+    const target = event.target as HTMLInputElement;
+    this.value = parseInt(target.value);
+    this.valueChanged.emit(this.value);
+    this.updateValuePercent();
+  }
+
+  updateValuePercent() {
+    if (this.inputRef) {
+      const percent = ((this.value - this.min) / (this.max - this.min)) * 100;
+      this.inputRef.style.setProperty('--value-percent', `${percent}%`);
```

Comment:
This does not work, because: 
1) The CSS variable must be defined in the `:root {}`. 
2) When the function is updated, and the  `this.inputRef.style.setProperty()` is changed, the component does not re-render again, so the styles are not updated in real time.

Reply:
What does not work? I can't reproduce any issue.

Reply:
Whatever you want to achieve with this:  this.inputRef.style.setProperty('--value-percent', `${percent}%`); does not work. 
I assume you want a gradual change of colors effect? But it's not working.